File: entry.py
"""Small data container representing a journal entry.

Fields are plain Python types for easy serialization: ``date`` is a
string in YYYY-MM-DD format, ``content`` contains HTML, ``tags`` is a
list of short strings, and ``attachments`` holds dicts with filename
and raw bytes.
"""

from datetime import date


class Entry:
    """A journal entry.

    This class is used as an in-memory representation of an entry
    retrieved from or written to the database.
    """

    def __init__(self, id: int | None = None, entry_date: str | None = None, title: str = "", content: str = "", tags: list[str] | None = None, attachments: list[dict] | None = None):
        """Initialize a journal entry."""
        self.id = id
        if entry_date is not None:
            self.date = entry_date
        else:
            from datetime import date as _date
            self.date = str(_date.today())
        self.title = title
        self.content = content  # HTML string
        self.tags = tags or []  # list of str
        self.attachments = attachments or []  # list of dict {'filename': str, 'data': bytes}
        self._undo_stack: list[str] = []  # For rich text undo support
        # per-entry display metadata
        self.font_family: str | None = None
        self.font_size: int | None = None
        # last saved timestamp (ISO string)
        self.last_saved: str | None = None

File: MyJournalWPF/Models/Entry.cs
using System;
using System.Collections.Generic;

namespace MyJournalWPF.Models
{
    public class Entry
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public string Tags { get; set; } = string.Empty;
        public string? FontFamily { get; set; }
        public int? FontSize { get; set; }
        public DateTime? LastSaved { get; set; }
        
        // Encrypted fields (stored in DB)
        public byte[]? EncryptedTitle { get; set; }
        public byte[]? EncryptedContent { get; set; }
        public byte[]? EncryptedTags { get; set; }
        
        public virtual ICollection<Attachment> Attachments { get; set; } = new List<Attachment>();
    }

    public class Attachment
    {
        public int Id { get; set; }
        public int EntryId { get; set; }
        public string FileName { get; set; } = string.Empty;
        public byte[] EncryptedData { get; set; } = Array.Empty<byte>();
        
        public virtual Entry? Entry { get; set; }
    }

    public class Config
    {
        public int Id { get; set; }
        public byte[] Salt { get; set; } = Array.Empty<byte>();
        public byte[] EncryptedTotpSecret { get; set; } = Array.Empty<byte>();
    }
}
